lib/cc/analyzer/engines_config_builder.rb

```diff
@@ -3,6 +3,19 @@
 module CC
   module Analyzer
     class EnginesConfigBuilder
+      class RegistryAdapter < SimpleDelegator
+        # Calling this is guarded by Registry#key?(name) so we can assume
+        # metadata itself will be present. We own the YAML loaded into the
+        # registry, so we can also assume the "channels" key will be present. We
+        # can't assume it will have a key for the given channel, but the nil
+        # value for the returned image key will trigger the desired error
+        # handling.
+        def fetch(name, channel)
+          metadata = self[name]
```

Comment:
I don't see any wins either way. If you prefer dropping the temp, that's fine with me. The name comes in handy in the comment that describes the assumptions so I'd have to reword that too.